sdks/python/apache_beam/transforms/stats.py

```diff
@@ -0,0 +1,216 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""This module has all statistic related transforms."""
+
+from __future__ import absolute_import
+from __future__ import division
+
+import heapq
+import math
+import sys
+from builtins import round
+
+import mmh3
+
+from apache_beam import coders
+from apache_beam.transforms.core import *
+from apache_beam.transforms.ptransform import PTransform
+
+__all__ = [
+    'ApproximateUniqueGlobally',
+    'ApproximateUniquePerKey',
+]
+
+
+class ApproximateUniqueGlobally(PTransform):
+  """
+  Hashes input elements and uses those to extrapolate the size of the entire
+  set of hash values by assuming the rest of the hash values are as densely
+  distributed as the sample space.
+
+  Args:
+    **kwargs: Accepts a single named argument "size" or "error".
+    size: an int not smaller than 16, which we would use to estimate
+      number of unique values.
+    error: max estimation error, which is a float between 0.01
+      and 0.50. If error is given, size will be calculated from error with
+      _get_sample_size_from_est_error function.
+  """
+
+  _NO_VALUE_ERR_MSG = 'Either size or error should be set. Received {}.'
+  _MULTI_VALUE_ERR_MSG = 'Either size or error should be set. ' \
+                         'Received {size = %s, error = %s}.'
+  _INPUT_SIZE_ERR_MSG = 'ApproximateUnique needs a size >= 16 for an error ' \
+                        '<= 0.50. In general, the estimation error is about ' \
+                        '2 / sqrt(sample_size). Received {size = %s}.'
+  _INPUT_ERROR_ERR_MSG = 'ApproximateUnique needs an estimation error ' \
+                         'between 0.01 and 0.50. Received {error = %s}.'
+
+  def __init__(self, size=None, error=None):
+
+    if None not in (size, error):
+      raise ValueError(self._MULTI_VALUE_ERR_MSG % (size, error))
+    elif size is None and error is None:
+      raise ValueError(self._NO_VALUE_ERR_MSG)
+    elif size is not None:
+      if not isinstance(size, int) or size < 16:
+        raise ValueError(self._INPUT_SIZE_ERR_MSG % (size))
+      else:
+        self._sample_size = size
+        self._max_est_err = None
+    else:
+      if error < 0.01 or error > 0.5:
+        raise ValueError(self._INPUT_ERROR_ERR_MSG % (error))
+      else:
+        self._sample_size = self._get_sample_size_from_est_error(error)
+        self._max_est_err = error
+
+  def expand(self, pcoll):
+    coder = coders.registry.get_coder(pcoll)
+    return pcoll \
+           | 'CountGlobalUniqueValues' \
+           >> (CombineGlobally(ApproximateUniqueCombineFn(self._sample_size,
+                                                          coder)))
+
+  def display_data(self):
```

Comment:
Sorry about the confusion. Composite transforms cannot contain display data. Can you add it to the combinefn?

Reply:
Thanks for catching it. It was fixed at most recent commit.